### PR TITLE
Feat/44 create a detailed product api endpoint for fetching individual product details

### PR DESCRIPTION
**Pull request by Ngha-Boris**

### CODE REVIEW

File: .hintrc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "extends": [
+    "development"
+  ],
+  "hints": {
+    "typescript-config/consistent-casing": "off"
+  }
+}
```
Comment (stephane-segning) [inline .hintrc:8]: Remove this please

File: package.json [hunk 2]
```diff
@@ -32,8 +32,9 @@
     "@prisma/client": "^5.14.0",
     "@t3-oss/env-nextjs": "^0.10.1",
     "@tanstack/react-query": "^5.50.0",
+    "@testing-library/react-hooks": "^8.0.1",
```
Comment (stephane-segning) [inline package.json:35]: Why?

File: package.json [hunk 3]
```diff
@@ -32,8 +32,9 @@
...
     "@trpc/client": "^11.0.0-rc.446",
-    "@trpc/react-query": "^11.0.0-rc.446",
+    "@trpc/react-query": "^10.45.2",
```
Comment (stephane-segning) [inline package.json:37]: Why changing the version?

**Review by stephane-segning — CHANGES_REQUESTED** [review]

File: package.json [hunk 4]
```diff
@@ -78,6 +78,7 @@
     "husky": "^9.1.6",
     "jest": "^29.7.0",
     "jest-environment-jsdom": "^29.7.0",
+    "msw": "^2.6.4",
```
Comment (stephane-segning) [inline package.json:81]: Remove this please

File: yarn.lock
Comment (stephane-segning) [inline yarn.lock:1]: Why did this file changed?

**Review by stephane-segning — CHANGES_REQUESTED** [review]

File: yarn.lock
Comment (Ngha-Boris) [reply yarn.lock:1]: Sir @stephane-segning I am also sure is because I removed some libraries in the package.json, like for example the msw library

**Review by stephane-segning — APPROVED** [review]

**Review by mbunwe-victor — COMMENTED** [review]
Good @Ngha-Boris

**Review by mbunwe-victor — APPROVED** [review]
Good @Ngha-Boris

**Review by bryandino673 — APPROVED** [review]

**Pull request merged**